src/aatoolbox/datasources/fewsnet/fewsnet.py

```diff
@@ -27,223 +33,335 @@
+        Country configuration
     """
-    if region_name not in _VALID_REGION_NAMES:
-        raise ValueError(f"Invalid region name {region_name}")
-    if region_code not in _VALID_REGION_CODES:
-        raise ValueError(f"Invalid region code {region_code}")
-
-    # convert to datetime if str
-    if not isinstance(date_pub, date):
-        date_pub = datetime.strptime(date_pub, _DATE_FORMAT)
-    # convert to path object if str
-    if isinstance(output_dir, str):
-        output_dir = Path(output_dir)
-
-    # we prefer the country data as this more nicely structured
-    # thus first check if that is available
-    country_path = _download_fewsnet_country(
-        date_pub=date_pub,
-        iso2=iso2,
-        output_dir=output_dir,
-        use_cache=use_cache,
-    )
-    if country_path is not None:
-        return country_path
-    else:
-        region_path = _download_fewsnet_region(
-            date_pub=date_pub,
-            region_name=region_name,
-            region_code=region_code,
-            output_dir=output_dir,
-            use_cache=use_cache,
+
+    def __init__(
+        self,
+        country_config,
+    ):
+        super().__init__(
+            country_config=country_config,
+            datasource_base_dir="fewsnet",
+            is_public=True,
+            # FN data can be regional, and thus we save the raw data
+            # in "glb" instead of in the iso3 folder, even though part of it is
+            # at country level
+            is_global_raw=True,
+            is_global_processed=False,
         )
-    if region_path is not None:
-        return region_path
-    else:
-        raise RuntimeError(f"No data found for {date_pub.strftime('%Y-%m')}")
 
+        self._iso2 = Country.get_iso2_from_iso3(self._country_config.iso3)
+        if self._iso2 is None:
+            raise KeyError(
+                "No ISO2 found for the given ISO3. Check your ISO3."
+            )
 
-def _download_fewsnet_country(
-    date_pub: date,
-    iso2: str,
-    output_dir: Path,
-    use_cache: bool,
-) -> Optional[Path]:
-    """
-    Download fewsnet data that covers the iso2 country.
+        if self._country_config.fewsnet is None:
+            raise KeyError(
+                "The country configuration file does not contain"
+                "any region name. Please update and try again."
+            )
 
-    Parameters
-    ----------
-    date_pub : date
-        date for which the data should be downloaded
-        only the year and month part are used
-        this commonly refers to the month of the Current Situation period
-    iso2 : str
-        iso2 code for which the data should be downloaded
-    output_dir: Path
-        path of dir to which the zip content should be written
-    use_cache: bool
-        if True, don't download if output_dir already exists
+        self._region_name = self._country_config.fewsnet.region_name
+        self._region_code = self._country_config.fewsnet.region_code
```

Comment:
I dont understand what you mean by this.. I mean `self._region_name` doesnt exist yet right? Or do you mean that I should call them later on in the code with `self._country_config.fewsnet.region_name` instead of `self._region_name`?

Reply:
I mean the latter, call them later with `self._country_config.fewsnet.region_name`

Reply:
Discussed bilaterally and will be improved with the custom config PR, so change after that is merged